code-challenges.test.js

```diff
@@ -13,61 +12,135 @@
+      friday: 15,
+      saturday: 65
+    }
+//     // Expected output: [10, 15, 20, 45, 60, 65, 100]
+    expect(leastToGreatest(studyMinutesWeek1)).toEqual([15, 15, 20, 30, 30, 60, 90])
+    expect(leastToGreatest(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+
+  })
+
+})
+// Failed succesfully: ReferenceError: leastToGreatest is not defined 
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// name: leastToGreatest
+// input: takes in an object
+// output: returns an array of the values sorted from least to greatest
+// process: 
+// - First I'm going to create a function named leastToGreatest and put object as a parameter
+// - inside of the function I will use the method object.values() so it will give me an array of the values inside the keys in the object and ill store that in a variable. then I'll use the sort method and put in it the parameter so it understand that i want the sorting to be from smallest to biggest  and store the sorted array in another variable and return that variable
+const leastToGreatest = (object) => {
+  const arrayValues = Object.values(object)
+  const arrayOrder = arrayValues.sort((a, b) => (a - b))
+  return arrayOrder
+}
+
+// test passed
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe("cummulativeSum", () =>{
+  it("returns an array of the accumulating sum", () =>{
+    const accountTransactions1 = [100, -17, -23, -9]
+//     // Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+    const accountTransactions2 = [250, -89, 100, -96]
+//     // Expected output: [250, 161, 261, 165]
 
-const accountTransactions3 = []
-// Expected output: []
+    const accountTransactions3 = []
+//     // Expected output: []
+    expect(cummulativeSum(accountTransactions1)).toEqual([100, -17, -23, -9])
+    expect(cummulativeSum(accountTransactions2)).toEqual([250, -89, 100, -96])
+    expect(cummulativeSum(accountTransactions3)).toEqual([])
+  })
+})
+// Failed successfully: ReferenceError: cummulativeSum is not defined
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// name: cummulativeSum
+// input: an array
+// output: returns an array of the accumulating sum
+// process: 
+// - first I'll create the function and call it cummulativeSum, then I'll put array as a parameter
+// - since the return is another array of the same lenght ill be using .map() and ill be putting inside of the map function the array parameter and also the index so I can access the indexes of the array and add them 
+const cummulativeSum = (array) => {
+  return array.map((array, index) => array[index - 0] + array[index]
```

Comment:
The trick with this one is to keep track of the accumulating sum throughout the iteration.
This is one approach you could use:
`let sum = 0`
`return array.map(num => sum += num)`

We use a variable sum that starts at 0. Then we use .map to iterate, and each iteration, the sum is added to the current number, then the variable is set to the new sum, which is returned in the new array.